packages/grafana-ui/src/components/TagsInput/TagItem.tsx

```diff
@@ -36,21 +36,35 @@ const getStyles = stylesFactory(({ theme, name }: { theme: GrafanaTheme; name: s
     nameStyle: css`
       margin-right: 3px;
     `,
+
+    buttonStyles: css`
+      border: none;
+      background: none;
+      padding: 0;
+    `,
   };
 });
 
 /**
  * @internal
  * Only used internally by TagsInput
  * */
-export const TagItem: FC<Props> = ({ name, onRemove }) => {
+export const TagItem: FC<Props> = ({ name, disabled, onRemove }) => {
   const theme = useTheme();
   const styles = getStyles({ theme, name });
 
   return (
     <div className={styles.itemStyle}>
       <span className={styles.nameStyle}>{name}</span>
-      <Icon className="pointer" name="times" onClick={() => onRemove(name)} />
+      <button
```

Comment:
Could we use `IconButton` here and apply custom styles to it?

Reply:
changed it to use `IconButton` in 9ba3ecd5ea37585d060a4540eff3fe2839a42157

i'm not overriding the button styles as i don't think it's a good practice in terms of maintenance in this case given overriding (disabling in this case) hover bg animations relies on the component implementation details that could easily change in the future and can break these overrides.

in case we don't want the new "look" for the remove button then i would revert the commit and not use the IconButton per the reason above. wdyt?

Reply:
or maybe we should have a primitive for a clickable icon in the first place? (IconButton provides extra functionality that aren't always needed and we use bare `Icon` with click handler quite extensively in the codebase)

Reply:
I don't think adjusting `IconButton` styles for this case is a bad idea, that's what `className` prop is there after all. I'm not sure what needs to be overridden for the `IconButton` here, but it seems to me that it's mostly the hover styles? Don't think we'd have any issues with that in the future. Additionally I'm not sure using Icon with `onClick` handler is good accessibility-wise.

Reply:
by "primitive for a clickable icon" i meant a simple button wrapping an icon, `Icon`s can already have `onClick`, making it accessible by removing it from the icon is the reason we made changes in this component. in various places we use `onClick` directly on the `Icon` because `IconButton` comes with extra styles and logic that do not fit those components use cases. This is basically the same reason why inside `IconButton` we do not use the `Button` component.

on top of that, while i agree that overriding styles is a viable solution most of the times, the hover animation here is achieved by creating a `before` pseudo element, i personally think that overriding components styles should be done cautiously, especially when targeting child/pseudo elements, and should ideally be limited to layouting.

that said, i'm going to remove hover styles by overriding them as this would probably need more extensive discussion, just wanted to post it here for dogfooding.